dnf/cli/commands/updateinfo.py

```diff
@@ -145,49 +144,23 @@ def run(self):
         else:
             self.display_summary(apkg_adv_insts, description)
 
-    def get_installed_version(self, name, arch):
-        if self._ina2evr_cache is None:
-            self._ina2evr_cache = {(pkg.name, pkg.arch): pkg.evr
-                                   for pkg in self.base.sack.query().installed()}
-        return self._ina2evr_cache.get((name, arch), None)
-
-    def compare_with_installed(self, apackage):
-        """Compare installed version with apackage version. Returns None either if
-        apackage is not installed, or if there are no update filters for apackage"""
-        key = (apackage.name, apackage.evr, apackage.arch)
-        if key not in self._update_filters_cmp_cache:
-            ievr = self.get_installed_version(key[0], key[2])
-            if ievr is None:
-                self._update_filters_cmp_cache[key] = None
-            else:
-                q = self.base.sack.query().filterm(name=key[0], evr=key[1])
-                if len(self.base._merge_update_filters(q, warning=False)) == 0:
-                    self._update_filters_cmp_cache[key] = None
-                else:
-                    self._update_filters_cmp_cache[key] = self.base.sack.evr_cmp(ievr, key[1])
-        return self._update_filters_cmp_cache[key]
-
-    def _older_installed(self, apackage):
-        """Test whether an older version of a package is installed."""
-        cmp = self.compare_with_installed(apackage)
-        if cmp is None:
-            return False
+    def _in_security_filters(self, apackage):
+        """If dnf security options used it returns True if present or security selection unused,
+        False if absent in security filter query"""
+        if self.base._update_security_filters:
+            key = (apackage.name, apackage.evr, apackage.arch)
+            q = self.base.sack.query().filterm(name=key[0], evr__gte=key[1])
```

Comment:
there is no need to build "key" tuple, you can use apackage.name and apackate.evr in filterm directly.